connect/runtime/src/test/java/org/apache/kafka/connect/util/clusters/EmbeddedConnectCluster.java

```diff
@@ -90,26 +132,48 @@ public void start() throws IOException {
      * Clean up any temp directories created locally.
      */
     public void stop() {
-        for (Connect worker : this.connectCluster) {
-            try {
-                worker.stop();
-            } catch (Exception e) {
-                log.error("Could not stop connect", e);
-                throw new RuntimeException("Could not stop worker", e);
-            }
-        }
-
+        connectCluster.forEach(this::stopWorker);
         try {
             kafkaCluster.after();
         } catch (Exception e) {
             log.error("Could not stop kafka", e);
             throw new RuntimeException("Could not stop brokers", e);
+        } finally {
+            Exit.resetExitProcedure();
+            Exit.resetHaltProcedure();
+        }
+    }
+
+    /**
+     * Provision and start an additional worker to the Connect cluster.
+     */
+    public void addWorker() {
+        connectCluster.add(new ConnectDistributed().startConnect(workerProps));
+    }
+
+    /**
+     * Decommission a worker from this Connect cluster.
+     */
+    public void removeWorker() {
+        if (connectCluster.isEmpty()) {
+            throw new IllegalStateException("Cannot remove worker. Cluster is empty");
+        }
+        Connect worker = connectCluster.remove(connectCluster.size() - 1);
+        stopWorker(worker);
+    }
+
+    private void stopWorker(Connect worker) {
+        try {
+            worker.stop();
+        } catch (Exception e) {
+            log.error("Could not stop connect", e);
+            throw new RuntimeException("Could not stop worker", e);
         }
     }
 
     @SuppressWarnings("deprecation")
     public void startConnect() {
-        log.info("Starting Connect cluster with {} workers. clusterName {}", connectCluster.length, connectClusterName);
+        log.info("Starting Connect cluster with {} workers. clusterName {}", numInitialWorkers, connectClusterName);
```

Comment:
Would it simplify the log message to instead:
```suggestion
        log.info("Starting Connect cluster '{}' with {} workers", connectClusterName, numInitialWorkers);
```

Reply:
That was a leftover intermediate change. Thanks